src/main/java/com/endava/internship/collections/StudentList.java

```diff
@@ -1,140 +1,393 @@
 package com.endava.internship.collections;
 
+
 import java.util.Collection;
 import java.util.Iterator;
 import java.util.List;
+import java.util.Objects;
+import java.util.Arrays;
 import java.util.ListIterator;
 
+
 public class StudentList implements List<Student> {
+    private int capacity;
+    private int size;
+    private Student[] array;
+
+
+    {
+        size = 0;
+    }
+    public StudentList() {
+        this.capacity = 10;
+        this.array = new Student[this.capacity];
+    }
+
+    public StudentList(int initialCapacity) {
+        if (initialCapacity >= 0) {
+            this.capacity = initialCapacity;
+            this.array = new Student[this.capacity];
+        } else {
+            if (initialCapacity < 0) {
+                throw new IllegalArgumentException("Initial capacity must be greater than 0");
+            }
+
+            this.capacity = 10;
+            this.array = new Student[this.capacity];
+        }
+
+    }
+
+    public StudentList(Collection<? extends Student> collection) {
+        if (!Objects.nonNull(collection)) {
+            throw new IllegalArgumentException("Collection does not exists");
+        } else {
+            this.size = collection.size();
+            this.capacity = this.size * 2;
+            this.array = new Student[this.capacity];
+            final Iterator iterator = collection.iterator();
 
+            while(iterator.hasNext()) {
+                Student t = (Student)iterator.next();
+                this.add(t);
+            }
+
+        }
+    }
     @Override
     public int size() {
-        //TODO
-        return 0;
+        return this.size;
     }
 
     @Override
     public boolean isEmpty() {
-        //TODO
-        return false;
+        return this.size == 0;
     }
 
     @Override
     public boolean contains(Object o) {
-        //TODO
-        return false;
+        boolean flag = false;
+        if(Objects.nonNull(o)){
+            for(int i = 0; i < this.size; i++) {
+                if (o.equals(this.array[i])) {
```

Comment:
In this case, I mean, that it will be better to change ordering of values and remove `Objects.nonNull()`.
Examples: 
`null.equals(something) - NullPointerException`
`something.equals(null) - false`

Reply:
fixed!